src/state.js

```diff
@@ -5,14 +5,26 @@ import DailyModel from "./models/dailyModel";
 
 import {challenge, hints, languages} from "../content/songs.json";
 import {Daily} from "./boilerplate/interval";
+import normalizeUrl from 'normalize-url';
 
 const MAX_ATTEMPTS = 6;
 
 let state;
 
 export const Mode = { Time: "timed", Free: "free" }
 
-const YOUTUBE_REGEX = /^((?:https?:)?\/\/)?((?:www|m)\.)?((?:youtube\.com|youtu.be))(\/(?:[\w\-]+\?v=|embed\/|v\/)?)([\w\-]+)(\S+)?$/
+const YOUTUBE_REGEX = /^((?:https?:)?\/\/)?((?:www|m|music)\.)?((?:youtube\.com|youtu.be))(\/(?:[\w\-]+\?v=|embed\/|v\/)?)([\w\-]+)(\S+)?$/
+
+const hostMap = {
```

Comment:
you might not need this, instead you can tell the type if you just match the domain regex you already have for youtube.  So if you just have a regex per host map that should be fine.